subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -2658,4 +2658,8 @@ static bool tls_is_supported(int family, int type, int proto)
 	return false;
 }
 
-NET_SOCKET_REGISTER(tls, AF_UNSPEC, tls_is_supported, ztls_socket);
+BUILD_ASSERT(CONFIG_NET_SOCKETS_TLS_PRIORITY < CONFIG_NET_SOCKETS_PRIORITY_DEFAULT,
```

Comment:
Why this check, it is not obvious by looking the code. Perhaps a comment would be good to have here.

Reply:
Since TLS and regular sockets fall under the same address family, TLS needs to be processed first to "capture" socket calls with secure protocols. Everything else will fall under the regular socket implementation. I'll add a comment.